pkg/initializer/validator/validator_suite_test.go

```diff
@@ -0,0 +1,273 @@
+package validator_test
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/embed"
+	"github.com/gardener/etcd-backup-restore/pkg/etcdutil"
+	"github.com/gardener/etcd-backup-restore/pkg/snapshot/snapshotter"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	outputDir                  = "../../../test/output"
+	etcdDir                    = outputDir + "/default.etcd"
+	snapstoreDir               = outputDir + "/snapshotter.bkp"
+	etcdEndpoint               = "http://localhost:2379"
+	snapshotterDurationSeconds = 15
+	snapCount                  = 10
+	keyPrefix                  = "key-"
+	valuePrefix                = "val-"
+	keyFrom                    = 1
+)
+
+var (
+	etcd         *embed.Etcd
+	err          error
+	keyTo        int
+	endpoints    = []string{etcdEndpoint}
+	etcdRevision int64
+)
+
+// fileInfo holds file information such as file name and file path
+type fileInfo struct {
+	name string
+	path string
+}
+
+func TestValidator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Validator Suite")
+}
+
+var _ = SynchronizedBeforeSuite(func() []byte {
+	var (
+		data               []byte
+		errCh              = make(chan error)
+		populatorStopCh    = make(chan bool)
+		populatorStoppedCh = make(chan bool)
+		ssrStopCh          = make(chan struct{})
+	)
+
+	err = os.RemoveAll(outputDir)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	logger := logrus.New()
+
+	err = os.RemoveAll(etcdDir)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	etcd, err = startEmbeddedEtcd(etcdDir, logger)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	go populateEtcd(logger, endpoints, errCh, populatorStopCh, populatorStoppedCh)
+
+	go func() {
+		<-time.After(time.Duration(snapshotterDurationSeconds * time.Second))
+		close(populatorStopCh)
```

Comment:
Will you apply fix in https://github.com/gardener/etcd-backup-restore/commit/8c3836da512c3922397a34ced21ab07f2f710d35 here. Cuase your current implementation cause race condition with the test cases, where revision on last snapshot may not equal to `keyTo` in some cases.